### DISCUSSION TITLE
Sharing a single model between multiple apps

### DISCUSSION THREAD

**Discussion by thobson** [prisma/prisma#18249] [Q&A]
Hi all, 

I'm looking for some advice re: the best way to handle a micro service architecture.

In my case I have a multi-tenant platform and several micro services e.g. a DocumentService and UserService

Most services reference a "Tenant" e.g. a Document has a foreign key to the Tenant, as does a User. Otherwise there are no dependencies between the different services.

I'm thinking how best to model this. I can see three approaches:

1. Manage all the tables and the Prisma client in a standalone project then share the generated code between the different services. This makes it easy to manage and enforce the foreign keys but it pretty much defeats the purpose of an independent micro service architecture. This is my least preferred option.

2. Model each micro service independently with foreign keys to the "Tenant" model/table, which is duplicated across all services. At runtime this is fine but I'm not sure how well it works with the migration tools. What happens if 2 services are both trying to manage the Tenant table?

3. As above but use different schemas for each service i.e. multiple Tenant tables across different schemas. I'm thinking this may be the best option but of course we'd lose the global referential integrity. This is not really a concern for me as it's to be expected with a micro service architecture. The only downside is the complexity of managing multiple schemas.

Does anyone have any thoughts or experience to share?

Thanks

**Comment by nurul3101** [comment]
Hello @thobson 👋 

I have encountered users who go with approach 1 as you can easily create a private npm package which has the prisma models and generated types which can be imported into different repositories. This approach can be useful if you have a lot of shared functionality between your microservices.

For approach 3, you can maintain some level of independence while still allowing for shared functionality and easier management of relationships between entities. However, it does require additional management overhead in terms of creating and managing the different database schemas.

**Reply by nurul3101** [reply]
@thobson If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by thobson** [comment]
Apologies, I thought I had already closed it.
